rockchip64-edge: bump to 6.14

amazingfate opened the pull request:
# Description

- Rebase patches based on 6.14-rc2
- Update some wifi driver code because there is a cfg80211 api change since 6.14: https://github.com/torvalds/linux/commit/7a53af85d3bbdbe06cd47b81a6d99a04dc0a3963
- Add patch to remove pci_clean in tools because pci is moved to selftest: https://github.com/torvalds/linux/commit/e19bde2269ca3611156fd0c078a71af0b6956545

I just make to build success. I will test the kernel on boards later.

# How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration._

- [x] `./compile.sh kernel BOARD=rock-5-itx BRANCH=edge KERNEL_GIT=shallow`

# Checklist:

_Please delete options that are not relevant._

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

amazingfate commented:
Tested with:
- armsom sige7(rk3588)
- rock 3a(rk3568)

EvilOlaf commented:
Oh wow, so I was actually right removing this from the Makefile with my own tests. Just didn't know why. Awesome.

amazingfate commented:
> Oh wow, so I was actually right removing this from the Makefile with my own tests. Just didn't know why. Awesome.

I just sent a mail to the patch list: https://lore.kernel.org/all/20250217045910.102934-1-liujianfeng1994@gmail.com/
Hope we can get rid of this patch in our tree when 6.14 is finally released.

EvilOlaf commented:
Tested with Orange Pi 5: https://paste.next.armbian.com/miyigiseya

EvilOlaf commented on lib/functions/compilation/patch/drivers_network.sh:
```diff
@@ -620,7 +620,7 @@ driver_rtl8723DS() {
 
 		# Attach to specific commit (was "branch:master")
-		local rtl8723dsver='commit:4b8d758f42ce38b75f9a98a083906b0d59d244c3' # Commit date: Dec 23, 2024 (please update when updating commit ref)
+		local rtl8723dsver='commit:ac64712f6767565bc3560e6ae412b780b07bc84c' # Commit date: Feb 16, 2024 (please update when updating commit ref)
```
2025 as well I assume?

amazingfate replied:
yes, I will fix it.

amazingfate commented:
After introducing vop mode improvment patch for hdmi1, all rk3588 boards that have only one hdmi port enabled are broken. I'm discussing with the patch author to see which way to fix:
https://lore.kernel.org/all/20250218033846.1251897-1-liujianfeng1994@gmail.com/

palachzzz commented:
BTW, bcmdhd-dkms is failing to build with that.

EvilOlaf commented:
> BTW, bcmdhd-dkms is failing to build with that.

logs?

palachzzz commented:
```
curl --data-binary output/logs/log-build-2d6e6b31-d707-4485-86ac-3c2d5a96f79c.log.ans https://paste.armbian.com/log
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot POST /log</pre>
</body>
</html>
```
Is there any highlight why it is so?

(need to remove .log at the end)
[log-build-2d6e6b31-d707-4485-86ac-3c2d5a96f79c.log.ans.log](https://github.com/user-attachments/files/18905580/log-build-2d6e6b31-d707-4485-86ac-3c2d5a96f79c.log.ans.log)

EvilOlaf commented:
.com paste seems broken. Try .de or .eu

EvilOlaf commented:
Can you try to get hold of the log file mentioned? ` /var/lib/dkms/bcmdhd-sdio/101.10.591.52.27-3/build/make.log `

rpardini commented:
>> Can you try to get hold of the log file mentioned? `/var/lib/dkms/bcmdhd-sdio/101.10.591.52.27-3/build/make.log`
> It seems it is cleaned up after failure

Yes, it dies with the rest of the chroot. I've added long ago code to dump the contents of the dkms log, as you can see here

https://github.com/armbian/build/blob/main/extensions/bcmdhd.sh#L43-L45

but seems it is simply not working. I will try to reproduce your build and fix it

rpardini commented:
> seems it is simply not working

Funny, it works on my tree: https://paste.next.armbian.com/quladowoca
(by "it works" I mean the dkms logs with the failures are included)
maybe I forgot to PR something, let me check

rpardini commented:
Either way the error is

```
   wl_cfg80211.c:12042:25: error: initialization of 'int (*)(struct wiphy *, struct wireless_dev *, unsigned int,  int *)' from incompatible pointer type 's32 (*)(struct wiphy *, struct wireless_dev *, s32 *)' {aka 'int (*)(struct wiphy *, struct wireless_dev *, int *)'} [-Werror=incompatible-pointer-types]
   12042 |         .get_tx_power = wl_cfg80211_get_tx_power,
         |                         ^~~~~~~~~~~~~~~~~~~~~~~~
   wl_cfg80211.c:12042:25: note: (near initialization for 'wl_cfg80211_ops.get_tx_power')
```

seems to me, "more of the same" churn on wifi over every release.

rpardini commented:
> (by "it works" I mean the dkms logs with the failures are included)
> maybe I forgot to PR something, let me check

Nah, I PR'ed it... https://github.com/armbian/build/pull/7653
@palachzzz can you build with `DEBUG=yes` and send the logs please?

iav commented:
trying to build with clang:
```
drivers/gpu/drm/panel/panel-simple-dsi.c:310:59: error: passing 'const u32 *' (aka 'const unsigned int *') to parameter of type 'u32 *' (aka 'unsigned int *') discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
           ret = of_get_drm_display_mode(panel->dev->of_node, mode, &p->desc->bus_format,
                                                                    ^~~~~~~~~~~~~~~~~~~~
   ./include/drm/drm_modes.h:511:45: note: passing argument to parameter 'bus_flags' here
                               struct drm_display_mode *dmode, u32 *bus_flags,
                                                                    ^
```
full log https://paste.next.armbian.com/ebeyunilul

iav commented:
no, same error https://paste.next.armbian.com/ovocohokoy

rpardini commented:
> no, same error [paste.next.armbian.com/ovocohokoy](https://paste.next.armbian.com/ovocohokoy)

I would start with clang on 0-patch, pure mainline kernels (eg `uefi-x86`, some others). rockchip64 is definitely not a prime clang target _yet_.

qbisi commented:
> trying to build with clang:
> 
> ```
> drivers/gpu/drm/panel/panel-simple-dsi.c:310:59: error: passing 'const u32 *' (aka 'const unsigned int *') to parameter of type 'u32 *' (aka 'unsigned int *') discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
>            ret = of_get_drm_display_mode(panel->dev->of_node, mode, &p->desc->bus_format,
>                                                                     ^~~~~~~~~~~~~~~~~~~~
>    ./include/drm/drm_modes.h:511:45: note: passing argument to parameter 'bus_flags' here
>                                struct drm_display_mode *dmode, u32 *bus_flags,
>                                                                     ^
> ```
> 
> full log https://paste.next.armbian.com/ebeyunilul

The patch file `general-add-panel-simple-dsi.patch` defines stuct simple-panel p with a const struct desc
```
struct panel_simple {
	struct drm_panel base;
	struct mipi_dsi_device *dsi;
	bool prepared;
	bool enabled;
	bool power_invert;

	struct device *dev;
	const struct panel_desc *desc;

	struct regulator *supply;

	struct gpio_desc *enable_gpio;
	struct gpio_desc *reset_gpio;
	int cmd_type;

	struct panel_cmds *on_cmds;
	struct panel_cmds *off_cmds;
	struct device_node *np_crtc;
	
	int reset_level;
	enum drm_panel_orientation orientation;
};
```

while the required parameter points to bus_flags should be a normal pointer as define in

https://github.com/torvalds/linux/blob/27102b38b8ca7ffb1622f27bcb41475d121fb67f/drivers/gpu/drm/drm_modes.c#L1156

maybe we should change `const struct panel_desc *desc` to `struct panel_desc *desc`?

SuperKali commented:
uhm i tried to compile it for rock 5 itx:

```bash
[    0.164611] /vop@fdd90000: Fixed dependency cycle(s) with /hdmi@fdea0000
[    0.164636] /hdmi@fdea0000: Fixed dependency cycle(s) with /vop@fdd90000
[    0.166400] /hdmi@fdea0000: Fixed dependency cycle(s) with /hdmi1-con
[    0.166423] /hdmi1-con: Fixed dependency cycle(s) with /hdmi@fdea0000
[    2.503223] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[    2.914965] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[    2.927678] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[    3.076666] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[    3.125567] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[    3.843825] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[    4.178635] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[    4.507224] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[    4.824945] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   12.767905] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   12.768151] platform fdea0000.hdmi: deferred probe pending: (reason unknown)
[   14.190803] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.251211] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.265456] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.267372] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.299607] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.324159] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.340144] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.351385] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.363150] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.368029] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.369023] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.372393] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.376712] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
[   14.434403] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
```

amazingfate commented:
> uhm i tried to compile it for rock 5 itx:
> 
> ```bash
> [    0.164611] /vop@fdd90000: Fixed dependency cycle(s) with /hdmi@fdea0000
> [    0.164636] /hdmi@fdea0000: Fixed dependency cycle(s) with /vop@fdd90000
> [    0.166400] /hdmi@fdea0000: Fixed dependency cycle(s) with /hdmi1-con
> [    0.166423] /hdmi1-con: Fixed dependency cycle(s) with /hdmi@fdea0000
> [    2.503223] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [    2.914965] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [    2.927678] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [    3.076666] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [    3.125567] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [    3.843825] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [    4.178635] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [    4.507224] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [    4.824945] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   12.767905] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   12.768151] platform fdea0000.hdmi: deferred probe pending: (reason unknown)
> [   14.190803] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.251211] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.265456] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.267372] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.299607] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.324159] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.340144] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.351385] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.363150] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.368029] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.369023] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.372393] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.376712] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> [   14.434403] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> ```

There is a necessary patch: https://www.spinics.net/lists/kernel/msg5564550.html, I will add it later.

SuperKali commented:
> > uhm i tried to compile it for rock 5 itx:
> > ```shell
> > [    0.164611] /vop@fdd90000: Fixed dependency cycle(s) with /hdmi@fdea0000
> > [    0.164636] /hdmi@fdea0000: Fixed dependency cycle(s) with /vop@fdd90000
> > [    0.166400] /hdmi@fdea0000: Fixed dependency cycle(s) with /hdmi1-con
> > [    0.166423] /hdmi1-con: Fixed dependency cycle(s) with /hdmi@fdea0000
> > [    2.503223] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [    2.914965] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [    2.927678] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [    3.076666] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [    3.125567] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [    3.843825] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [    4.178635] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [    4.507224] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [    4.824945] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   12.767905] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   12.768151] platform fdea0000.hdmi: deferred probe pending: (reason unknown)
> > [   14.190803] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.251211] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.265456] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.267372] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.299607] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.324159] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.340144] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.351385] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.363150] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.368029] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.369023] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.372393] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.376712] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > [   14.434403] rockchip-drm display-subsystem: [drm] *ERROR* failed to get pll_hdmiphy0
> > ```
> 
> There is a necessary patch: https://www.spinics.net/lists/kernel/msg5564550.html, I will add it later.

I applied manually the patch and it works 💯 

![image](https://github.com/user-attachments/assets/a7364890-d8bb-4abc-87d6-91afe17db69a)

palachzzz commented:
The same with OPi 5 Max, applied proposed patch, and HDMI finally got back with no issues in dmesg

paolosabatino commented:
Quick check against a rk3318 tv box and rk3399 Orange PI 4 LTS, both boards boot and dmesg is clean is enough :+1:

SuperKali reviewed (approved):
All work as expected 😊

EvilOlaf reviewed (approved):
Re-tested Orangepi5 with full Gnome desktop on Bookworm with backported mesa. Works great!

palachzzz commented:
Re-tested with 6.14-rc4, works fine, including bcmdhd.
Is the understanding when it could be merged?
